presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -151,6 +154,23 @@ public TableStatistics getTableStatistics(
         }
     }
 
+    private TableStatistics createZeroStatistics(Map<String, ColumnHandle> columns, Map<String, Type> columnTypes)
+    {
+        TableStatistics.Builder result = TableStatistics.builder();
+        result.setRowCount(Estimate.of(0));
+        columns.forEach((columnName, columnHandle) -> {
+            Type columnType = requireNonNull(columnTypes.get(columnName), "columnType is null");
+            ColumnStatistics.Builder columnStatistics = ColumnStatistics.builder();
+            columnStatistics.setNullsFraction(Estimate.of(0));
```

Comment:
Why not create `ColumnStatistics.ZERO`, so that next time, we don't have to think what should be the null fraction?
(as long as you conditionally also set data size, you can have sth like `ColumnStatistics.zero()` returning a builder)

Reply:
> I thought about this, but i didn't want to have data size for the columns that we don't add data size for.
That shouldn't matter since it's only for non-null rows. I would use `ColumnStatistics.ZERO`. Otherwise we do something here which is not needed which only adds complexity to the code.

Reply:
That will simplify the code for a bit, but not much. But that won't be consistent to what we return for non empty and empty partitions.